ops/terraform/services/static-site/data-sources.tf

```diff
@@ -0,0 +1,227 @@
+    }
+
+    actions = ["s3:GetObject"] ##, "s3:ListBucket"]
+    resources = [
+      "${aws_s3_bucket.static_site.arn}/*"
+    ]
+  }
+
+  statement {
+    sid    = "JenkinsRWObjects"
+    effect = "Allow"
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+    actions = ["s3:*"]
+    resources = [
+      "${aws_s3_bucket.static_site.arn}",
+      "${aws_s3_bucket.static_site.arn}/*"
+    ]
+    condition {
+      test     = "ArnEquals"
+      variable = "AWS:UserId"
+      values   = ["arn:aws:iam::${local.account_id}:role/cloudbees-jenkins"]
+    }
+  }
+
+  statement {
+    sid = "AllowKMSAccess"
+    actions = [
+      "s3:GetObject",
+      "s3:ListBucket"
+    ]
+    resources = [
+      "${aws_s3_bucket.static_site.arn}",
+      "${aws_s3_bucket.static_site.arn}/*"
+    ]
+
+    principals {
+      type        = "AWS"
+      identifiers = [aws_cloudfront_origin_access_identity.static_site_identity.iam_arn]
+    }
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SecureTransport"
+      values   = ["true"]
+    }
+  }
+
+  statement {
+    sid     = "OnlySecureTransport"
+    actions = ["s3:*"]
+    condition {
+      test     = "Bool"
+      variable = "aws:SecureTransport"
+      values   = ["false"]
+    }
+    effect = "Deny"
+    principals {
+      type        = "*"
+      identifiers = ["*"]
+    }
+    resources = [
+      "${aws_s3_bucket.static_site.arn}",
+      "${aws_s3_bucket.static_site.arn}/*"
+    ]
+  }
+}
+
+
+data "aws_iam_policy_document" "cloudfront_log_policy" {
+  depends_on = [aws_s3_bucket.cloudfront_logging, aws_cloudfront_distribution.static_site_distribution]
+  statement {
+    sid = "AllowCloudFrontServicePrincipal"
+
+    effect = "Allow"
+    principals {
+      type        = "Service"
+      identifiers = ["cloudfront.amazonaws.com"]
+    }
+    actions = ["s3:*"] # ["s3:PutObject"]
+    resources = [
+      "${aws_s3_bucket.cloudfront_logging.arn}",
+      "${aws_s3_bucket.cloudfront_logging.arn}/*"
+    ]
+    # condition {
+    #   test = "ForAnyValue:StringEquals"
+    #   variable = "AWS:SourceAccount" 
+    #   values = [data.aws_caller_identity.current.account_id]
+    # }
+    condition {
+      test     = "StringEquals"
+      variable = "AWS:SourceArn"
+      values   = [aws_cloudfront_distribution.static_site_distribution.arn]
+    }
+  }
+  statement {
+    sid     = "OnlySecureTransport"
+    actions = ["s3:*"]
+    condition {
+      test     = "Bool"
+      variable = "aws:SecureTransport"
+      values   = ["false"]
+    }
+    effect = "Deny"
+    principals {
+      type        = "*"
+      identifiers = ["*"]
+    }
+    resources = [
+      "${aws_s3_bucket.cloudfront_logging.arn}",
+      "${aws_s3_bucket.cloudfront_logging.arn}/*"
+    ]
+  }
+}
+
+# data "aws_acm_certificate" "env_issued" {
+#   domain      = "${local.static_site_fqdn}"
+#   statuses    = ["ISSUED"]
+#   types       = ["IMPORTED"]
+#   most_recent = true
+# }
```

Comment:
I think this may need an explicit TODO comment. Otherwise, it should be removed.